### PR TITLE
G-691 | Adding call back for On-Confirm

### PR DESCRIPTION
**Pull request by Nexengineer**

### CODE REVIEW

File: src/In.ProjectEKA.HipService/User/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        public AcceptedResult LinkFor(JObject request)
+        {
+            backgroundJob.Enqueue(() => LinkPatient(request));
+            return Accepted();
+        }
+        
+        [NonAction]
+        public async Task LinkPatient(JObject request)
+        {
+            Console.Write(request);
```
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/User/UserController.cs:33]: @Nexengineer Console huh? Use ILogger\<> what we do in CareContext controller.

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/User/UserController.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5922534&bid=19405494)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/User/UserController.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5922534&bid=19405494)

File: src/In.ProjectEKA.HipService/User/UserController.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json.Linq;
+
+    [ApiController]
+    public class UserController : ControllerBase
+    {
+        private readonly IBackgroundJobClient backgroundJob;
+        private readonly GatewayClient gatewayClient;
+        private readonly ILogger<UserController> logger;
```
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/User/UserController.cs:17]: Codacy found an issue: [Remove unassigned field 'logger', or set its value.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5922534)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/User/UserController.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5922534&bid=19405494)

File: src/In.ProjectEKA.HipService/User/UserController.cs (hunk 1 above)
Comment (Nexengineer) [reply src/In.ProjectEKA.HipService/User/UserController.cs]: done

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/User/UserController.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5922534&bid=19405494)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/User/UserController.cs  2
- src/In.ProjectEKA.HipService/User/AuthOnConfirm.cs  1
- src/In.ProjectEKA.HipService/User/AuthOnConfirmResponse.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5922534&bid=19405494)

**Pull request merged**